include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp

```diff
@@ -0,0 +1,194 @@
+    bool is_positive() const { return measure > 0.0; }
+};
+
+template <>
+struct distance_measure<long double>
+    : public distance_measure_floating<long double> {};
+
+template <>
+struct distance_measure<double>
+    : public distance_measure_floating<double> {};
+
+template <>
+struct distance_measure<float>
+    : public distance_measure_floating<float> {};
+
+} // detail
+
+
+namespace detail_dispatch
+{
+
+// TODO: this is effectively a strategy, but for internal usage.
+// It might be moved to the strategies folder.
+
+template <typename CalculationType, typename CsTag>
+struct get_distance_measure
+        : not_implemented<CsTag>
+{};
+
+template <typename CalculationType>
+struct get_distance_measure<CalculationType, cartesian_tag>
+{
+    typedef detail::distance_measure<CalculationType> result_type;
+
+    template <typename SegmentPoint, typename Point>
+    static result_type apply(SegmentPoint const& p1, SegmentPoint const& p2,
+                             Point const& p)
+    {
+        typedef CalculationType ct;
+
+        // Construct a line in general form (ax + by + c = 0),
+        // (will be replaced by a general_form structure in next PR)
+        ct const x1 = geometry::get<0>(p1);
+        ct const y1 = geometry::get<1>(p1);
+        ct const x2 = geometry::get<0>(p2);
+        ct const y2 = geometry::get<1>(p2);
+        ct const a = y1 - y2;
+        ct const b = x2 - x1;
+        ct const c = -a * x1 - b * y1;
+
+        // Returns a distance measure
+        // https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line#Line_defined_by_an_equation
+        // dividing by sqrt(a*a+b*b) is not necessary for this distance measure,
+        // it is not a real distance and purpose is to detect small differences
+        // in collinearity
+        result_type result;
+        result.measure = a * geometry::get<0>(p) + b * geometry::get<1>(p) + c;
+
+        return result;
+    }
+};
+
+template <typename CalculationType>
+struct get_distance_measure<CalculationType, spherical_tag>
+{
+    typedef detail::distance_measure<CalculationType> result_type;
+
+    template <typename SegmentPoint, typename Point>
+    static result_type apply(SegmentPoint const& , SegmentPoint const& ,
+                             Point const& )
+    {
+        // TODO, optional
+        result_type result;
+        return result;
+    }
+};
+
+template <typename CalculationType>
+struct get_distance_measure<CalculationType, spherical_equatorial_tag>
+{
+    typedef detail::distance_measure<CalculationType> result_type;
+
+    template <typename SegmentPoint, typename Point>
+    static result_type apply(SegmentPoint const& , SegmentPoint const& ,
+                             Point const& )
+    {
+        // TODO, optional
+        result_type result;
+        return result;
+    }
+};
+template <typename CalculationType>
+struct get_distance_measure<CalculationType, geographic_tag>
```

Comment:
This specialization is the same as the one for `spherical_tag` so it could be implemented as:

    template <typename CalculationType>
    struct get_distance_measure<CalculationType, geographic_tag>
        : get_distance_measure<CalculationType, spherical_tag>
    {};

Reply:
OK - will do this and merge, thank you